DirectProgramming/DPC++FPGA/Tutorials/Features/private_copies/README.md

```diff
@@ -0,0 +1,194 @@
+ 
+ ### In Third-Party Integrated Development Environments (IDEs)
+
+You can compile and run this tutorial in the Eclipse* IDE (in Linux*) and the Visual Studio* IDE (in Windows*). For instructions, refer to the following link: [Intel® oneAPI DPC++ FPGA Workflows on Third-Party IDEs](https://software.intel.com/en-us/articles/intel-oneapi-dpcpp-fpga-workflow-on-ide)
+
+## Examining the Reports
+Locate `report.html` in the `private_copies_report.prj/reports/` or `private_copies_s10_pac_report.prj/reports/` directory. Open the report in any of Chrome*, Firefox*, Edge*, or Internet Explorer*.
+
+On the main report page, scroll down to the section titled "Estimated Resource Usage". Each kernel name ends in the `private_copies` attribute argument used for that kernel, e.g., `kernelCompute1` uses a `private_copies` attribute value of 1. You can verify that the number of RAMs used for each kernel increases with the `private_copies` value used, with the exception of `private_copies` 0. Using `private_copies` 0 instructs the compiler to choose a default value, which is often close to the value that would give you maximum throughput.
+
+## Running the Sample
+
+ 1. Run the sample on the FPGA emulator (the kernel executes on the CPU):
+     ```
+     ./private_copies.fpga_emu     (Linux)
+     private_copies.fpga_emu.exe   (Windows)
+     ```
+2. Run the sample on the FPGA device:
+     ```
+     ./private_copies.fpga         (Linux)
+     ```
+
+
+### Example of Output
+```
+PASSED_fpga_compile
+Kernel time when private_copies is set to 0: 1441.45 ms
+Kernel throughput when private_copies is set to 0: 0.568 GFlops
+Kernel time when private_copies is set to 1: 2916.440 ms
+Kernel throughput when private_copies is set to 1: 0.281 GFlops
+Kernel time when private_copies is set to 2: 1458.260 ms
+Kernel throughput when private_copies is set to 2: 0.562 GFlops
+Kernel time when private_copies is set to 3: 1441.450 ms
+Kernel throughput when private_copies is set to 3: 0.568 GFlops
+Kernel time when private_copies is set to 4: 1441.448 ms
+Kernel throughput when private_copies is set to 4: 0.568 GFlops
+PASSED: The results are correct
+```
+
+### Discussion of Results
+
+The stdout output shows the throughput (GFlops) for each kernel. 
+
+When run on the Intel® PAC with Intel Arria10® 10 GX FPGA hardware board, we see that the throughput of the kernel doubles when going from 1 to 2 private copies for array `a`. Further increasing the number of private copies does not increase the throughput achieved, i.e., increasing the number of private copies above 2 will spend additional RAM resources for no additional throughput gain. As such, for this tutorial design, maximal throughput is best achieved when using 2 private copies. The small jump in throughput observed from using 3 private copies instead of 2 can be attributed to the third private copy allowing consequent outer loop iterations to launch slightly sooner then they would with 2 private copies.
```

Comment:
This is much better!  Just a bit of a tweak needed here.

Your statement 'Further increasing the number of private copies does not increase the throughput achieved' is not correct, which you acknowledge later.  I find this a bit confusing.  How about this:

When run on the Intel® PAC with Intel Arria10® 10 GX FPGA hardware board, we see that the throughput of the kernel approximately doubles when going from 1 to 2 private copies for array `a`. Increasing to 3 private copies has a very small affect, and further increasing the number of private copies does not increase the throughput achieved. This means that increasing private copies beyond 2 will incur an area penalty for little or no throughput gain. As such, for this tutorial design, optimal throughput/area tradeoff is achieved when using 2 private copies.  This is as expected based on an analysis of the code, as there are two inner loops which are able to run in parallel if they each have their own private copy of the array. The small jump in throughput observed from using 3 private copies instead of 2 can be attributed to the third private copy allowing subsequent outer loop iterations to launch slightly sooner then they would with 2 private copies due to internal delays in tracking the use of each private copy.